src/clj/game/core-actions.clj

```diff
@@ -57,27 +57,31 @@
   (let [c (update-in card [:zone] #(map to-keyword %))
         last-zone (last (:zone c))
         src (name-zone (:side c) (:zone c))
-        from-str (if (nil? src) nil (str " from their " src))
-        label (if (and (not (= last-zone :play-area))
-                       (not (and (= (:side c)  "Runner") (= last-zone :hand) (= server "Grip")))
-                       (or (and (= (:side c)  "Runner") (not (:facedown c)))
-                           (:rezzed c)
+        from-str (when-not (nil? src) (str " from their " src))
+        label (if (and (not= last-zone :play-area)
+                       (not (and (= (:side c)  "Runner")
+                                 (= last-zone :hand)
+                                 (= server "Grip")))
+                       (or (and (= (:side c)  "Runner")
+                                (not (:facedown c)))
+                           (rezzed? c)
                            (:seen c)
                            (= last-zone :deck)))
-                (:title c) "a card")
+                (:title c)
+                "a card")
```

Comment:
the whole naming in here just asks to be replaced by an appropriate `card-str` call, maybe modified slightly. Ideally all card naming, if not very specialized for some reason, could be handled in that one place.